keps/754-time-dependent-job-priority/README.md

```diff
@@ -0,0 +1,188 @@
+This functionality aims to prioritize the execution of jobs that have been waiting 
+for an extended period, even if they have a lower initial priority.  
+By introducing time-dependent priority adjustments, the system can ensure that jobs 
+that have been deferred receive the attention they require.  
+This KEP outlines the necessary changes and considerations to incorporate this 
+time-based priority mechanism.
+
+## Motivation
+
+Lower-priority jobs sometimes get postponed for long time than expected.  
+To address situations where jobs remain unexecuted even after a certain period of time,
+we implementate this KEP.
+
+### Goals
+
+Provide a mechanism where the priority of a job increase after a certain period of
+time has elapsed.
+
+### Non-Goals
+
+The timed out job is going to the head of the ClusterQueue instead of changing 
+the priority.
+
+## Proposal
+
+Replace the priority of Jobs with a calculation that takes into account the time since the Job was created.  
+This replacement will modify the calculation methods for Preemption and Queueing strategy.  
+The priority of Job that don't have `WorkloadPriorityClass` will remain unchanged.
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+### User Stories
+
+#### Story 1
+
+The job doesn't have a high priority but is intended for production, so it needs to start execution within three hours.
+
+
+### Risks and Mitigations
+
+If the job is just popped out for scheduling, the running job would be preempted easily by other Jobs.  
+That's why this mechanism should be based on priority of job.
+
+## Design Details
+
+### Kueue WorkloadPriorityClass API
+
+We introduce the Kueue `WorkloadPriorityClass` API.
+
+```golang
+// WorkloadPriorityClass defines basePriority and time dependent calculation logic
+type WorkloadPriorityClass struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   WorkloadPrioritySpec   `json:"spec,omitempty"`  
+}
+
+type WorkloadPrioritySpec struct {
+  // basePriority is initial priority of workload
+	BasePriority int32 `json:"basePriority,omitempty"`
+
+  // stepPriority is a value that is added to the basePriority every time the duration defined by delayForStep elapses. 
+  StepPriority int32 `json:"stepPriority,omitempty"`
+
+  // maxPriority defines the maximum number of Priority. We don't want it to go to infinity.
+  maxPriority int32 `json:"maxPriority,omitempty"`
+
+  // delayForStep indicates the time to wait before changing the priority after a timeout
+  DeleyForStep time.Duration `json:"delayForStep,omitempty"`
```

Comment:
can you make all of these optional?

Reply:
maybe it should also something like this:

```yaml
spec:
  priority: 0
  dynamicPriorityPolicy: # optional
    step: 1
    max: 100
    delayForStep: 1m
```

Reply:
Thanks for your comments!
I agree. Fixed it.